cmake/flash/CMakeLists.txt

```diff
@@ -1,48 +1,61 @@
-get_property(ENV_VARS GLOBAL PROPERTY FLASH_SCRIPT_ENV_VARS)
+assert_not(FLASH_SCRIPT "FLASH_SCRIPT has been removed; use BOARD_FLASH_RUNNER")
+assert_not(DEBUG_SCRIPT "DEBUG_SCRIPT has been removed; use BOARD_DEBUG_RUNNER")
 
-set(ENV_VARS_FORMATTED "")
-foreach(env_var ${ENV_VARS})
-  list(APPEND ENV_VARS_FORMATTED
-    ${env_var}=${${env_var}}
-    )
-endforeach()
-
-list(APPEND ENV_VARS_FORMATTED
-                O=${PROJECT_BINARY_DIR}
-  KERNEL_ELF_NAME=${KERNEL_ELF_NAME}
-  KERNEL_HEX_NAME=${KERNEL_HEX_NAME}
-  KERNEL_BIN_NAME=${KERNEL_BIN_NAME}
-        BOARD_DIR=${BOARD_DIR}
-              GDB=${CMAKE_GDB}
-          OPENOCD_DEFAULT_PATH=${OPENOCD_DEFAULT_PATH}
-          OPENOCD=${OPENOCD}
+set(RUNNER_ARGS_COMMON
+  # Required:
+  "--board-dir=${BOARD_DIR}"
+  "--kernel-elf=${KERNEL_ELF_NAME}"
```

Comment:
Why "kernel-*"? Wouldn't it be better to call it "zephyr-*"? This immediately confused me.

Reply:
The variable KERNEL_ELF_NAME is already present in CMake before this series. I'm using the same name in Python for consistency. Do you think we should change it?

Reply:
Yes, actually I think we should. Especially given that in the future we might decide to split the kernel into a different image for security or conformance reasons.

Reply:
@carlescufi: I looked into this further, and that change would be user-visible, since cmake/app/boilerplate.cmake gets all of these names from CONFIG_KERNEL_BIN_NAME. That would need to become CONFIG_ZEPHYR_BIN_NAME for consistency, which would break out of tree configs.

Given that and since this naming scheme is pre-existing, I'd like to regard this as out of scope for this series if OK with you.

Reply:
@mbolivar I understand. I will create a separate issue to deal with this, thanks for looking into it.